galsim/photon_array.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+import numpy as np
+from ._galsim import PhotonArray
+import galsim
+
+class FRatioAngles(object):
+    """A surface-layer operator that assigns photon directions based on the f/ratio and
+    obscuration.
+
+    Assigns arrival directions at the focal plane for photons, drawing from a uniform
+    brightness distribution between the obscuration angle and the angle of the FOV defined
+    by the f/ratio of the telescope.  The angles are expressed in terms of slopes dx/dz
+    and dy/dz.
+
+    @param fratio           The f-ratio of the telescope (e.g. 1.2 for LSST)
+    @param obscuration      Linear dimension of central obscuration as fraction of aperture
+                            linear dimension. [0., 1.).  [default: 0.0]
+    @param rng              A random number generator to use or None, in which case an rng
+                            will be automatically constructed for you. [default: None]
+    """
+    def __init__(self, fratio, obscuration=0.0, rng=None):
+
+        if fratio < 0:
+            raise ValueError("The f-ratio must be positive.")
+        if obscuration < 0 or obscuration >= 1:
+            raise ValueError("The obscuration fraction must be between 0 and 1.")
+        if rng is None:
+            ud = galsim.UniformDeviate()
```

Comment:
This line is currently uncovered by the tests.  We should have a test that makes FRatioAngles without a specified seed.  Probably also one that uses an existing BaseDeviate rather than an integer seed.  We don't necessarily need to do the full test of the distribution with these, especially the default one, but tests of min/max values are probably worthwhile.